packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPickerUtils/useSvgParser.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'react';
+
+import { traverseSvg } from './SvgParserHelper';
+
+interface IConvertComponentProps {
+  selectedColor: string;
+  iconString: string;
+}
+
+export const useSvgParser = ({ selectedColor, iconString }: IConvertComponentProps) => {
+  const [svgCode, setSvgCode] = useState<string>('');
+
+  useEffect(() => {
+    try {
+      const decodeBase64 = (str: string) => {
+        if (typeof atob === 'function') {
+          return atob(str);
+        } else {
+          return Buffer.from(str, 'base64').toString('binary');
+        }
+      };
+
+      // Extract the SVG element from the base64 encoded string
+      const svgString = decodeBase64(iconString!);
+      const parser = new DOMParser();
+      const svgDocument = parser.parseFromString(svgString, 'image/svg+xml');
+      const svgRoot = svgDocument.documentElement;
```

Comment:
Interesting. I didn't know the DOM parser was exposed by the browser API! Cool!

Reply:
Yup. Thanks :)